theano/gpuarray/opt.py

```diff
@@ -1075,15 +1074,14 @@ def local_gpua_careduce(op, context_name, inputs, outputs):
             # be None
             reshaped_gpu_inputs = [gpu_reshaped_x]
             if greduce.supports_c_code(reshaped_gpu_inputs):
-                reduce_reshaped_x = host_from_gpu(
-                    greduce(gpu_reshaped_x))
+                reduce_reshaped_x = greduce(gpu_reshaped_x)
 
                 if reduce_reshaped_x.ndim != outputs[0].ndim:
                     out_shp = []
                     for i in range(x.ndim):
                         if i not in op.axis:
                             out_shp.append(shape_i(x, i))
-                    unreshaped_reduce = reduce_reshaped_x.reshape(
+                    unreshaped_reduce = GpuReshape(len(out_shp))(
```

Comment:
What about:

GpuReshape(len(out_shp))(reduce_reshaped_x, tensor.stack(out_shp))

I forgot to add the reduce_reshaped_x parameter